### ISSUE TITLE
Narrow search results

### ISSUE THREAD

**Issue by mpollmeier** [syohex/emacs-helm-ag#187]
I'm a former vim user and loved QFGrep (https://github.com/sk1418/QFGrep), which let's you `grep` and `grep -v` on the results. I.e. with `<leader> g` you can specify a filter, and with `<leader> v` you can specify an inverse filter (what not to show). 

Is there any way to narrow down the results in helm-ag or with standard emacs tools? I haven't found anything :(

![QFGrep](https://camo.githubusercontent.com/3afcaaa19ebda45dbbabe288c063460b7d247a7f/68747470733a2f2f7261772e6769746875622e636f6d2f736b313431382f7368617265645265736f75726365732f6d61737465722f5146477265702f64656d6f2e676966)

**Comment by syohex** [comment]
Input `-v` option directory as below.

![helm-ag](https://cloud.githubusercontent.com/assets/554281/11434430/aad2269e-950f-11e5-9510-011d7e7ccc85.gif)


Or write following command.
```lisp
(defun helm-do-ag-inverse ()
  (interactive)
  (let ((helm-ag-command-option "-v"))
    (call-interactively 'helm-do-ag)))
```

**Comment by mpollmeier** [comment]
Thanks for your answer, but that's not really what I was looking for. Let me rephrase:
When I use helm-ag to search, it displays a buffer with all the results. That's great, but often I want to further filter down these results. E.g. for all these results, exclude any result that contains 'apple'. And then, out of the remaining results, only show those which contain 'green'.  

Does that make sense?

**Comment by syohex** [comment]
Sorry I misunderstood.

In `helm-ag`, you can use `!pattern`(You need space preceding pattern if you use it as first pattern).

![helm-ag](https://cloud.githubusercontent.com/assets/554281/11449478/8e780602-95ba-11e5-948e-c76dbaa6b0e3.gif)

In `helm-do-ag`, you can use negative look ahead pattern like `^(?!.*pattern).+$`.(It is difficult to input such pattern :-()

**Comment by mpollmeier** [comment]
awesome, that's what i was looking for! thank you!
